token/filetoken/filetoken.go

```diff
@@ -0,0 +1,114 @@
+package filetoken
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/user"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/puppetlabs/pe-sdk-go/log"
+	"github.com/puppetlabs/pe-sdk-go/token"
+)
+
+// fileToken struct
+type fileToken struct {
+	path string
+}
+
+// NewFileToken constructs a new filetoken
+func NewFileToken(path string) token.Token {
+	fileToken := fileToken{path: getPath(path)}
+	return &fileToken
+}
+
+// Read reads the token from a file and returns a valid token.
+// If the token is not valid an error will be returned
+func (ft *fileToken) Read() (string, error) {
+	return ft.readToken()
+}
+
+//DeleteFile deletes the token file
+//If the token is not valid, or file is a symlink an error will be returned
+func (ft *fileToken) Delete() error {
+	_, err := ft.readToken()
+	if err != nil {
+		return err
+	}
+	return ft.deleteFile()
```

Comment:
`readToken` will return an error if `token` is not valid, in that case we won't delete the file, is this expected?